x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -753,13 +761,23 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     geometry: Geometry | Position[],
     defaultFields: Record<string, Record<string, string>>
   ) {
-    const indexPattern = await this.getIndexPattern();
-    await addFeatureToIndex(indexPattern.title, geometry, this.getGeoFieldName(), defaultFields);
+    const indexList = await this.getSourceIndexList();
+    if (indexList.length !== 1) {
+      throw new Error(
```

Comment:
I think the actions that call these add/delete methods need to handle the exception and display a toast when an exception is thrown.